### PR TITLE
Copyediting to improve style and language of overview section

### PR DESCRIPTION
**Pull request by NickAnderegg**
Copyediting/revision of four files in the overview section to improve overall style and language quality.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -18,18 +18,20 @@ As a Platform as a Service, or PaaS, Platform.sh automatically manages everythin
 
 Platform.sh covers not only all of your hosting needs but also most of your DevOps needs. It is a simple, single tool that covers the application life-cycle from development to production and scaling.
 
-You only need to write your code, including a few YAML files that specify your desired infrastructure, commit it to Git, and push.  You don't need to setup anything manually. The web server is already setup and configured, as are any database, search engine or cache that you specify.
+You only need to write your code, including a few YAML files that specify your desired infrastructure, commit it to Git, and push.  You don't need to setup anything manually. The web server is already setup and configured, as is any database, search engine or cache that you specify.
```
Comment (Crell) [inline src/README.md]: This one's subtle. It's a list of items (plural), but the items are singular.  So in aggregate is that singular or plural?  Is this a US/UK divide? :smile:

File: src/README.md [hunk 2]
```diff
@@ -18,18 +18,20 @@ As a Platform as a Service, or PaaS, Platform.sh automatically manages everythin
...
 
-Every branch you push is a fully running environment &mdash; complete with your application code, a copy of your database, a copy of your search index, a copy of your user files, everything &mdash; and its automatically generated URL can be sent to stakeholders or to automated CI systems.  It really is "what would my site look like if I merged this to production?"  Every time.
+Every branch you push is a full copy of a production environment&mdash;complete with your application code, a copy of your database, a copy of your search index, a copy of your user files, everything&mdash;and its automatically generated URL can be sent to stakeholders or to automated CI systems.  It really is "what would my site look like if I merged this to production?"  Every time.
```
Comment (Crell) [inline src/README.md]: AFAIK mdash should be surrounded by spaces, no?

Also, the text change here alters the meaning.  A branch becomes a copy of production only once, when created.  It can then diverse dramatically.  The key point is that it is a fully self-contained site, not an alternate view on production data.  I think this text change is incorrect.

File: src/README.md [hunk 3]
```diff
@@ -18,18 +18,20 @@ As a Platform as a Service, or PaaS, Platform.sh automatically manages everythin
...
-4. Fail-Proof Deployments: every time you test a new feature you also test the deployment process.
-5. Everything is build oriented with fully consistent and a repeatable build process, allowing you to easily keep your application updated and secure.
+1. **Unified Environment:** All of your services (MySQL, ElasticSearch, MongoDB, etc...) are managed inside the cluster and included in the price, with no external single-points-of-failure. When you backup an environment, you get a fully consistent snapshot of your whole application.
+2. **Multi-Services & Multi-App:** You can deploy multiple applications (for example, in a microservice-based architecture), using multiple data backends (MySQL, Postgres, Redis etc..) written in multiple frameworks (Drupal + NodeJS + Flask, for example) in multiple languages, all in the same cluster.
+3. **Full Cluster Cloning Technology:** The full production cluster can be cloned in under a minute&mdash;including all of its data&mdash;to create on-the-fly, ephemeral development environments that are a byte-level copy of production.
+4. **Fail-Proof Deployments:** Every time you test a new feature, you also test the deployment process.
```
Comment (Crell) [inline src/README.md]: I think the comma on this line is extraneous.  (And I'm normally a comma dealer.)

File: src/README.md (hunk 2 above)
Comment (NickAnderegg) [reply src/README.md]: Em-dashes should never be surrounded by spaces! People doing it like — this or like -- this is a weird style quirk.

How about "a fully independent environment" instead? "Fully running" comes off as extremely vague to me

File: src/overview/build-deploy.md [hunk 4]
```diff
@@ -9,36 +9,36 @@ Every time you push to a live branch (a git branch with an active environment at
...
+Interpreted languages like PHP or Node.js may not seem like they have to be compiled, but with modern package management tools like Composer or npm, as well as the growing use of CSS preprocessors such as Sass, most modern web applications need a "build" step between their source code and their production execution code.  At Platform.sh, we aim to make that easy.  The build step includes the entire application container&mdash;from language version to build tools to your code&mdash;rebuilt every time.
 
-That build process is under your control, and reruns on every Git push.  Every Git push is a validation not only of your code but of your build process.  Whether that's installing packages using Composer or Bundler, compiling TypeScript or Sass, or building your application code in Go or Java, your build process is vetted every time you push.
+That build process is under your control and runs on every Git push.  Every Git push is a validation not only of your code, but of your build process.  Whether that's installing packages using Composer or Bundler, compiling TypeScript or Sass, or building your application code in Go or Java, your build process is vetted every time you push.
```
Comment (Crell) [inline src/overview/build-deploy.md]: Disagree on the additional comma here.

File: src/overview/build-deploy.md [hunk 5]
```diff
@@ -9,36 +9,36 @@ Every time you push to a live branch (a git branch with an active environment at
...
+1. **The build step should be environment-independent.** This is paramount to ensure development environments are truly perfect copies of production. This means you can not connect to services (like the database) during the build.
+2. **The final built application must be read-only.** If your application requires writing to the filesystem, you can specify the directories that require Read/Write access. These should not be directories that have code, because that would be a security risk.
 
 ## Building the application
 
-After you push your code the first build step is to validate your configuration files (`.platform.app.yaml`, `.platform/services.yaml` and `.platform/routes.yaml`). The git server will issue an error if validation fails, and nothing will have happened on the server.
+After you push your code, the first build step is to validate your configuration files (i.e. `.platform.app.yaml`, `.platform/services.yaml`, and `.platform/routes.yaml`). The Git server will issue an error if validation fails, and nothing will happen on the server.
```
Comment (Crell) [inline src/overview/build-deploy.md]: First comma is extraneous.

File: src/overview/build-deploy.md (hunk 4 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: Contrasted coordinate elements should always be separated with a comma.

So it would be "a validation of your code and of your build process," but "a validation not only of your code, but of your build process." The construction "not only ..., but ..." is what makes the coordinate element require a comma.

File: src/overview/build-deploy.md [hunk 6]
```diff
@@ -9,36 +9,36 @@ Every time you push to a live branch (a git branch with an active environment at
...
+The live environment is composed of multiple containers&mbdash;both for your application(s) and for the services it depends on. It also has a virtual network connecting them, as well as a router to route incoming requests to the appropriate application.
 
 Based on your application type the system will select one of our pre-built container images and run the following:
 
-1. First, any dependencies specified in the `.platform.app.yaml` file are installed.  Those include tools like Sass, Gulp, Drupal Console, or many others that you may need.  
+1. First, any dependencies specified in the `.platform.app.yaml` file are installed.  Those include tools like Sass, Gulp, Drupal Console, or any others that you may need.  
 
-2. Second, depending on the “build flavor” specified in the configuration file we run a series of standard commands. The default for PHP containers, for example, is simply to run `composer install`.
+2. Then, depending on the “build flavor” specified in the configuration file, we run a series of standard commands. The default for PHP containers, for example, is simply to run `composer install`.
```
Comment (Crell) [inline src/overview/build-deploy.md]: I disagree with both changes on this line.  First should be followed by Second, and the comma is extraneous.

File: src/README.md (hunk 3 above)
Comment (NickAnderegg) [reply src/README.md]: Which comma?

File: src/overview/build-deploy.md [hunk 7]
```diff
@@ -9,36 +9,36 @@ Every time you push to a live branch (a git branch with an active environment at
...
 
-3. Finally, run the “build hook” from the configuration file.  The build hook is simply one or more shell commands that you write to finish creating your production code base.  That could be compiling Sass files, running a Gulp or Grunt script, rearranging files on disk, compiling an application in a compiled language, or whatever else you want.  Note that at this point all you have access to is the file system; there are no services or other databases available.
+3. Finally, we run the “build hook” from the configuration file.  The build hook comprises one or more shell commands that you write to finish creating your production code base.  That could be compiling Sass files, running a Gulp or Grunt script, rearranging files on disk, compiling an application in a compiled language, or whatever else you want.  Note that at this point, all you are able to access is the file system; there are no services or other databases available.
```
Comment (Crell) [inline src/overview/build-deploy.md]: The final comma is extraneous.

File: src/overview/build-deploy.md (hunk 5 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: Introductory clauses should always have a comma. Basically, if you can invert the sentence, it should have a comma.

Thus, "The first build step is to validate your configuration files after you push your code" requires a comma when it's inverted to "After you push your code, the first build step..."

File: src/overview/build-deploy.md (hunk 6 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: This is pretty much a preference thing, but on a list of three, I think it's stylistically better to use "First,...Then,...Finally." with "second" reserves for lists of four or more.

File: src/overview/pricing.md [hunk 8]
```diff
@@ -1,26 +1,26 @@
...
-* Support
+* multiple Backend services (MySQL, PostgreSQL, Elasticsearch, Redis, Solr..)
+* support
 
-You can switch between plans (downgrade or upgrade) freely but note that
+You can switch between plans (downgrade or upgrade) freely, but note that
 reducing storage is currently not supported for technical reasons. If you need
-to reduce storage please create a support ticket. You will always be billed the
+to reduce storage, please create a support ticket. You will always be billed the
 prorated rate of your plan over the period it was used.
```
Comment (Crell) [inline src/overview/pricing.md]: As long as we're editing this line let's move it to a single line, rather than a hard break in the middle of a sentence.

Comment (Crell) [reply src/overview/pricing.md]: I really thought I'd eliminated all of those but I keep finding more...

File: src/overview/build-deploy.md (hunk 6 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: Also, introductory words, phrases, and clauses get a comma. If you can lop off the words at the beginning of a sentence and it's not an descriptor of the word at the beginning of the sentence, it almost always requires a comma.

File: src/overview/pricing.md [hunk 9]
```diff
@@ -1,26 +1,26 @@
...
 
 You may cancel your plan at any time and you will only be billed for the actual
 period used.
```
Comment (Crell) [inline src/overview/pricing.md]: Ibid.  Single line while we're at it.

Comment (Crell) [reply src/overview/pricing.md]: (And in the rest of the file.  I won't note each place it's needed here.)

File: src/overview/build-deploy.md (hunk 7 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: Technically, I'm being lax on the comma here. I could go with "Note that, at this point, all you are..." to set off the non-essential clause, but since both phrases serve as an introductory clause, it's not totally necessary.

File: src/overview/pricing.md [hunk 10]
```diff
@@ -77,12 +77,12 @@ Production plans come in three flavors:
 
 ### Multiple Applications in a single project
 
-All Platform.sh plans support multiple applications in a single cluster but
-they all share the global resource of the cluster.
+All Platform.sh plans support multiple applications in a single cluster, but
+they share the global resources of the cluster.
```
Comment (Crell) [inline src/overview/pricing.md]: Single-line it, and I dispute this comma.

File: src/README.md (hunk 1 above)
Comment (NickAnderegg) [reply src/README.md]: It's the "or." If it were "and" then it would be "are," but "or" gets "is." Also, there's a missing oxford comma I forgot to add.

File: src/overview/pricing.md (hunk 10 above)
Comment (NickAnderegg) [reply src/overview/pricing.md]: It's two independent clauses separated by a coordinating conjunction. There is not a single style guide that would call that comma incorrect.

File: src/overview/structure.md [hunk 11]
```diff
@@ -1,13 +1,21 @@
 # Structure
 
-Every application you deploy on Platform.sh is built as a *virtual cluster*, containing a series of containers.  The master branch of your Git repository is always deployed as a production cluster.  Any other branch can be deployed as a development cluster.  By default you can have up to three live development clusters at once but you can buy more on a per-project basis.
+Every application you deploy on Platform.sh is built as a *virtual cluster*, containing a set of containers.  The master branch of your Git repository is always deployed as the production cluster.  Any other branch can be deployed as a development cluster.  By default, you can have up to three live development clusters at once, but you can buy more on a per-project basis.
 
-There are three types of containers within your cluster: One *Router*, one or more *Applications*, and zero or more *Services*.  When deployed, they conceptually look like this:
+There are three types of containers within your cluster:
+- one *Router*
+- one or more *Application* containers
+- zero or more *Service* containers
```
Comment (Crell) [inline src/overview/structure.md]: By convention bulleted lists always use *, not -.

Also, put a newline between the preceding line and the list.

File: src/overview/structure.md [hunk 12]
```diff
@@ -1,13 +1,21 @@
...
+When deployed, they conceptually look like this:
 
 ![](https://platform.sh/sites/default/files/scalable-images/architecture-simple.svg)
 
 
-All of those containers are managed by three special files in your Git repository: `.platform/routes.yaml`, `.platform/services.yaml`, and `.platform.app.yaml`.
+All of those containers are managed by three special files in your Git repository:
+- `.platform/routes.yaml`
+- `.platform/services.yaml`
+- `.platform.app.yaml`
```
Comment (Crell) [inline src/overview/structure.md]: See previous on lists.

File: src/overview/structure.md [hunk 13]
```diff
@@ -18,19 +26,25 @@ yourproject/
     services.yaml
     routes.yaml
   .platform.app.yaml
-  <your files here>
+  <your application code>
```
Comment (Crell) [inline src/overview/structure.md]: Let's go with "your application files", since it's technically not all code, and code implies inlining it which would of course be silly.

Comment (NickAnderegg) [reply src/overview/structure.md]: I think this was a case of brain and hands not agreeing on what to type, since "your application code" makes no sense.

File: src/overview/structure.md [hunk 14]
```diff
@@ -18,19 +26,25 @@ yourproject/
...
+There always must be one Application container in a cluster, but there may be more.
+
+Each Application container corresponds to a `.platform.app.yaml` file in the repository.  If there are 3 `.platform.app.yaml` files, there will be three Application containers.  Application containers hold the code you provide via your Git repository.  Application containers are always built off of one of the Platform.sh-provided language-specific images, such as “PHP 5.6”, “PHP 7.2”, or “Python 3.6”. It is also possible to have multiple Application containers running different languages or versions.
 
-For most typical applications there is only one `.platform.app.yaml` file, which is generally placed at the repository root.
+For many applications, there is only one `.platform.app.yaml` file, which is generally placed at the repository root.
```
Comment (Crell) [inline src/overview/structure.md]: Why this change?  Typical projects are single-app, so this is true.

**Comment by Crell** [comment]
Hooray for nitpicky wordsmithing!

File: src/README.md (hunk 1 above)
Comment (Crell) [reply src/README.md]: I'll go with that.  And Oxford Comma Forevar!

File: src/README.md (hunk 2 above)
Comment (Crell) [reply src/README.md]: I have never heard a specific call on mdashes like that.  Citation?

"Fully independent environment" works for me.

File: src/README.md (hunk 3 above)
Comment (Crell) [reply src/README.md]: "every time you test a new feature you also test the deployment process".  The comma after "feature" is IMO unnecessary.

File: src/overview/build-deploy.md (hunk 4 above)
Comment (Crell) [reply src/overview/build-deploy.md]: I"ll buy that.  Stet.

File: src/overview/build-deploy.md (hunk 5 above)
Comment (Crell) [reply src/overview/build-deploy.md]: Stet.

File: src/README.md (hunk 2 above)
Comment (NickAnderegg) [reply src/README.md]: It's a thing in some newspaper style guides to surround an em-dash with a space, but [APA style](http://blog.apastyle.org/apastyle/2010/09/computer-editing-tip-em-dashes.html), [MLA style](http://www.mlanet.org/page/mla-style-manual-punctuation), and [Chicago style](https://apvschicago.com/2011/05/em-dashes-and-ellipses-closed-or-spaced.html) all say no spaces.

I typically default to APA style since I've written literally hundreds of pages of APA-formatted papers, and it has heavy overlap with MLA in terms of stylistic choices.

File: src/overview/build-deploy.md (hunk 6 above)
Comment (Crell) [reply src/overview/build-deploy.md]: I was referring to the comma after "configuration file".  I agree that the Second/Then should have a comma.

File: src/README.md (hunk 3 above)
Comment (NickAnderegg) [reply src/README.md]: It wouldn't need a comma if it were "You also test the deployment process every time you test a new feature," but since "every time you test a new feature" precedes the subject of the sentence, it's functioning as an introductory phrase which should be set off by a comma.

File: src/overview/build-deploy.md (hunk 7 above)
Comment (Crell) [reply src/overview/build-deploy.md]: I would prefer both commas then, as the core sentence is "all you are able to..."; Both "Note that" and "at this point" are additions, so let's either go with both commas or none.

File: src/overview/pricing.md (hunk 10 above)
Comment (Crell) [reply src/overview/pricing.md]: Stet.

File: src/overview/build-deploy.md (hunk 6 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: "depending on the 'build flavor' specified in the configuration file" is a subordinate clause. Comma!

File: src/overview/pricing.md (hunk 9 above)
Comment (NickAnderegg) [reply src/overview/pricing.md]: Done.

File: src/overview/pricing.md (hunk 8 above)
Comment (NickAnderegg) [reply src/overview/pricing.md]: Done.

File: src/overview/structure.md (hunk 14 above)
Comment (NickAnderegg) [reply src/overview/structure.md]: Fixed.

File: src/overview/structure.md (hunk 13 above)
Comment (NickAnderegg) [reply src/overview/structure.md]: Fixed.

File: src/overview/structure.md (hunk 12 above)
Comment (NickAnderegg) [reply src/overview/structure.md]: Fixed.

File: src/overview/structure.md (hunk 11 above)
Comment (NickAnderegg) [reply src/overview/structure.md]: Fixed.

File: src/overview/build-deploy.md (hunk 7 above)
Comment (NickAnderegg) [reply src/overview/build-deploy.md]: Fixed.

**Comment by NickAnderegg** [comment]
Alright, looks like all the comments have either been responded to or addressed in the copy itself.
